### DISCUSSION TITLE
I made a mistake in my question on a Homework Assessment, how do I reset students' variants?

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#6742] [Q&A] [answered]
If I make a mistake creating a question in a PrairieLearn assessment of type "Homework", and have a bunch of broken questions, what's the best strategy to fix it and is there a way to mark a variant as "broken"?  The students have already started on the assignment.

For context, I have three problematic questions:

1) A question where there's a minor spelling mistake in the `question.html` file.

2) A question where **not all** of the variants are broken, just a few variants don't have the correct calculation

3) A question where **all** of the variants are broken and students cannot answer the question.

**Comment by echuber2** [comment]
As pointed out by @firasm on Slack, @mwest1066 previously described these workarounds:

> I'm afraid that we don't have a great solution for this at the moment -- the "right" fix is #5480 as @ffund  pointed out, but that's still in the works.
> 
> For now there are two fixes, depending on whether some students have successfully completed the question already:
> 
> 1. If all students are stuck with broken variants:
>    Edit the question and change its `uuid` (in the `info.json` file) to a new UUID. This will make PL treat it as a completely new unrelated question, and reset it for all students. This will generate new variants from scratch for all students and wipe all existing records of this question (including any existing scores and attempts). 
> 2. If some students have successfully completed the question already, but a few students are stuck with broken variants:
>    Copy the question to a new question that has a different UUID, change the name to add "(version 2)" to the end of it, and add this new question to the homework as well. Then tell students that they should do either one of the two questions. You could even put the two questions together in their own zone with `"bestQuestions": 1`, so that they will be given points for one or the other but not both.

**Reply by firasm** [reply]
Just for completeness in this case:

> 1. A question where there's a minor spelling mistake in the question.html file.

You should just be able to edit the question.html file and the changes should show up live to all the students as soon as you Save and Sync.

**Comment by eliotwrobson** [comment]
Shameless plug for https://github.com/PrairieLearn/PrairieLearn/pull/6531, that feature would fix this issue (which comes up all the time).

**Comment by jonatanschroeder** [comment, marked as answer]
There is now a better alternative, described in the documentation: https://prairielearn.readthedocs.io/en/latest/faq/#i-updated-my-question-but-students-are-still-seeing-the-old-version-how-can-i-get-students-to-see-the-new-version
